### ISSUE TITLE
Linker fails in cpp_utils files while building

### ISSUE THREAD

**Issue by sulyokg** [nkolban/ESP32_Explorer#22]
Hi Neil!
I've tried to compile your Explorer app with Eclipse and ESP-IDF. Eclipse is configured based on your youtube tutorial. I've been experimenting with ESP32 for just a couple of days. I've copied Explorer source files under a subdirectory in my test application, and invoked from my main app.
My test project stucture:

```
- [project]
-     components - dir
-         esp32-ds18b20 - dir
-         esp32-owb - dir
-     main - dir
-         cpp_utils - dir
-         explorer - dir
-         adafruit - dir
-             Adafruit_GFX files
-             Adafruit_SSD1306 files
-         main_helloworld.cpp and some files
```

The compile seems to be successful, but linking is fail. I've copied first cpp_util directory to components dir with the same result. Without Explorer, the app is compiled and linked successfully with the ds18b20 library in the components directory, and adafruit libraries in the main/adafruit subdirectory.

Build log is here:
https://pastebin.com/raw/7Q80biQJ

Thanks.

**Comment by chegewara** [comment]
Yeah, there is a lot changes in cpp_utils library and we didnt check Explorer for quite some time. cpp_utils should be in component folder, and be sure there is component.mk in cpp_utils folder.

**Comment by sulyokg** [comment]
Yeah, i've tried with [project]/components/cpp_utils with component.mk in this subdir, but linking fails.

**Comment by chegewara** [comment]
Like i said, there was a lot changes in cpp_utils, we need to make some fix change in Explorer. Is it rush project?

**Comment by sulyokg** [comment]
I see, it's not urgent, I just wanted to try it out.

**Comment by chegewara** [comment]
@sulyokg is compilling with no issues, but i didnt check if it works. If you find any issues just let us know, thanks.

**Comment by chegewara** [comment]
Finaly i compiled it and all seems to work fine, i think so issue can be closed.
